azuredevops.go

```diff
@@ -276,42 +341,47 @@ func runAzureDevOpsHandler(c *gin.Context) {
 // @Param azureDevOpsPR body azureDevOpsPR _ "AzureDevOps PR "
 // @Param environment query string true "wharf build environment"
 // @Success 200 {object} wharfapi.ProjectRunResponse "OK"
-// @Failure 400 {object} string "Bad request"
-// @Failure 401 {object} string "Unauthorized or missing jwt token"
+// @Failure 400 {object} problem.Response "Bad request"
+// @Failure 401 {object} problem.Response "Unauthorized or missing jwt token"
 // @Router /azuredevops/triggers/{projectid}/pr/created [post]
 func prCreatedTriggerHandler(c *gin.Context) {
 	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
 
 	t := azureDevOpsPR{}
 	if err := c.BindJSON(&t); err != nil {
-		c.Error(err)
-		c.JSON(http.StatusBadRequest, err.Error())
+		ginutil.WriteInvalidBindError(c, err,
+			"One or more parameters failed to parse when reading the request body for pull request.")
 		return
 	}
 
 	if t.EventType != "git.pullrequest.created" {
-		c.JSON(
-			http.StatusBadRequest,
-			fmt.Sprintf("Expected git.pullrequest.created trigger, got: %s instead.", t.EventType))
+		err := fmt.Errorf("expected git.pullrequest.created trigger, got: %s", t.EventType)
+		ginutil.WriteProblemError(c, err, problem.Response{
+			Type: "prob/provider/azuredevops/invalid-event-type-error",
+			Title: "Invalid event type.",
+			Status: http.StatusBadRequest,
+			Detail: "Invalid event type.",
+		})
```

Comment:
The event type is probably valid, although it's unsupported. Suggest changing to something like `/prob/provider/azuredevops/unsupported-event-type`

Suggest to also extract the event type as a local const and then include the value in the detail as well.

```go
const eventTypePullRequest = "git.pullrequest.created"

// ...

Detail: fmt.Sprintf("Received event type %q, while only %q is supported.", t.EventType, eventTypePullRequest),
```